lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+
+    return Container(
+      width: MediaQuery.of(context).size.width,
+      constraints: const BoxConstraints(
+        minHeight: 44,
+        maxHeight: 124),
+      padding: const EdgeInsets.symmetric(horizontal: 14,vertical:11),
+      decoration: BoxDecoration(color: designVariables.bgSearchInput),
+      child: SingleChildScrollView(
+        controller: scrollController,
+        child: Row(children: [
+          Expanded(child: Wrap(
+            spacing: 6,
+            runSpacing: 4,
+            crossAxisAlignment: WrapCrossAlignment.center,
+            children: [
+              ...selectedUserIds.map((userId) => SelectedUserChip(userId: userId)),
+              _buildSearchField(designVariables, zulipLocalizations)]))])));
+  }
+
+  Widget _buildSearchField(DesignVariables designVariables,
+      ZulipLocalizations zulipLocalizations) => IntrinsicWidth(
+        child: SizedBox(
+          height: 22,
+          child: TextField(
+            controller: searchController,
+            cursorColor: designVariables.foreground,
+            style: const TextStyle(
+              fontSize: 17,
+              height: 1.0,
+            ),
+            scrollPadding: EdgeInsets.zero,
+            decoration: InputDecoration(
+              hintText: zulipLocalizations.newDmSheetSearchHint,
+              hintStyle: TextStyle(
+                fontSize: 17,
+                height: 1.0,
+                color: designVariables.textInput.withFadedAlpha(0.5)),
+              border: InputBorder.none,
+            ))));
+}
+
+class SelectedUserChip extends StatelessWidget {
+  const SelectedUserChip({
+    super.key,
+    required this.userId});
+
+  final int userId;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final user = store.users[userId]!;
+
+    return Container(
+      height: 22,
+      decoration: BoxDecoration(color: designVariables.bgMenuButtonSelected, borderRadius: BorderRadius.circular(3)),
+      child: Row(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Avatar(userId: userId, size: 22, borderRadius: 3),
+          Padding(
+            padding: const EdgeInsets.symmetric(horizontal: 5, vertical: 3),
+            child: Text(user.fullName,
+              style: TextStyle(
+                fontSize: 16,
+                height: 1.0,
+                color: designVariables.labelMenuButton)))]));
+  }
+}
+
+class NewDmUserList extends StatelessWidget {
+  const NewDmUserList({
+    super.key,
+    required this.filteredUsers,
+    required this.selectedUserIds,
+    required this.onUserSelected});
+
+  final List<User> filteredUsers;
+  final Set<int> selectedUserIds;
+  final void Function(int) onUserSelected;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return ListView.builder(
```

Comment:
There is an 8px top padding that applies to the scrollable:

![image](https://github.com/user-attachments/assets/6579b02b-5f6a-4c51-ac2f-5a8810f49933)

I assume that's something that allows you to scroll over. Perhaps `ListView.padding` will be helpful.